codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestDecoderTest.java

```diff
@@ -345,7 +345,70 @@ public void testHeaderWithNoValueAndMissingColon() {
 
         assertTrue(channel.writeInbound(Unpooled.copiedBuffer(requestStr, CharsetUtil.US_ASCII)));
         HttpRequest request = channel.readInbound();
-        System.err.println(request.headers().names().toString());
+        assertTrue(request.decoderResult().isFailure());
+        assertTrue(request.decoderResult().cause() instanceof IllegalArgumentException);
+        assertFalse(channel.finish());
+    }
+
+    @Test
+    public void testMultipleContentLengthHeaders() {
+        EmbeddedChannel channel = new EmbeddedChannel(new HttpRequestDecoder());
+        String requestStr = "GET /some/path HTTP/1.1\r\n" +
+                "Content-Length: 1\r\n" +
```

Comment:
Looking at the code, this is not an issue but you may want to test non-contiguous `content-length` headers:

```java
String requestStr = "GET /some/path HTTP/1.1\r\n" +
                 "Content-Length: 1\r\n" +
                 "Connection: close\r\n" +
                 "Content-Length: 0\r\n\r\n" +
                 "b";